File: Leetcode3379. Transformed array/Solution.java
class Solution {
    public int[] constructTransformedArray(int[] nums) {
        int n = nums.length;
        int[] res = new int[n];
        for(int i=0; i<n; i++) {
            res[i] = nums[(((i+nums[i])%n) + n) % n];
        }
        return res;
    }
}


// runtime - 0ms
import java.util.*;
import static java.lang.Math.*;
import static java.lang.System.*;

class Solution {
  static final int INT_MAX = (int) 1e9, INT_MIN = -INT_MAX;
  static final long LONG_MAX = (long) 1e14, LONG_MIN = -LONG_MAX;
  static final int mod = (int) 1e9 + 7;

  public int[] constructTransformedArray(int[] nums) {
    int n = nums.length;
    int[] res = new int[n];
    for (int i = 0; i < n; i++) {
      res[i] = nums[(((i + nums[i]) % n) + n) % n];
    }
    return res;
  }
}

File: Leetcode3379. Transformed array/Solution.py
class Solution:
    def constructTransformedArray(self, nums: List[int]) -> List[int]:
        res = [0] * len(nums)

        for i in range(len(nums)):
            index = (nums[i] + i) % len(nums)
            res[i] = nums[index]

        return res

# runtime - 27ms
class Solution:
    def constructTransformedArray(self, nums: List[int]) -> List[int]:
        n = len(nums)
        result = [0] * n

        for idx in range(n):
            destIdx = (idx + nums[idx]) % n
            if destIdx < 0:
                destIdx += n
            result[idx] = nums[destIdx]


        return result


# runtime - 35ms
class Solution:
    def constructTransformedArray(self, nums: List[int]) -> List[int]:
        n = len(nums)
        out = [0] * n
        for i in range(n):
            x = (i + nums[i])%n
            out[i] = nums[x]
        return out


# runtime - 38ms
class Solution:
    def constructTransformedArray(self, nums: List[int]) -> List[int]:
        n = len(nums)
        return [nums[(i + x) % n] for i, x in enumerate(nums)]
        
